programs/clearing_house/src/controller/position.rs

```diff
@@ -415,100 +415,67 @@ pub fn update_position_and_market(
-        )?;
-
-        let per_lp_delta_quote = get_proportion_i128(
-            delta.quote_asset_amount,
-            AMM_RESERVE_PRECISION,
-            total_lp_shares,
-        )?;
-
-        let lp_delta_base =
-            get_proportion_i128(per_lp_delta_base, user_lp_shares, AMM_RESERVE_PRECISION)?;
-        let lp_delta_quote =
-            get_proportion_i128(per_lp_delta_quote, user_lp_shares, AMM_RESERVE_PRECISION)?;
-
-        let per_lp_position_delta = PositionDelta {
-            base_asset_amount: -per_lp_delta_base,
-            quote_asset_amount: -per_lp_delta_quote,
-        };
-
-        update_amm_position(market, &per_lp_position_delta, true)?;
+    if user_lp_shares == 0 {
+        return Ok((0, 0, 0));
+    }
 
-        // 1/5 of fee auto goes to market
-        // the rest goes to lps/market proportional
-        let lp_fee = (fee_to_market - (fee_to_market / 5)) // todo: 80% retained
-            .checked_mul(cast_to_i128(user_lp_shares)?)
-            .ok_or_else(math_error!())?
-            .checked_div(cast_to_i128(total_lp_shares)?)
-            .ok_or_else(math_error!())?;
+    // update Market per lp position
+    let per_lp_delta_base = get_proportion_i128(
+        delta.base_asset_amount,
+        AMM_RESERVE_PRECISION,
+        total_lp_shares,
+    )?;
 
-        let per_lp_fee = if lp_fee > 0 {
-            lp_fee
-                .checked_mul(AMM_RESERVE_PRECISION_I128)
-                .ok_or_else(math_error!())?
-                .checked_div(cast_to_i128(user_lp_shares)?)
-                .ok_or_else(math_error!())?
-        } else {
-            0
-        };
+    let per_lp_delta_quote = get_proportion_i128(
+        delta.quote_asset_amount,
+        AMM_RESERVE_PRECISION,
+        total_lp_shares,
+    )?;
 
-        // update per lp position
-        market.amm.market_position_per_lp.quote_asset_amount = market
-            .amm
-            .market_position_per_lp
-            .quote_asset_amount
-            .checked_add(per_lp_fee)
-            .ok_or_else(math_error!())?;
+    let lp_delta_base =
+        get_proportion_i128(per_lp_delta_base, user_lp_shares, AMM_RESERVE_PRECISION)?;
+    let lp_delta_quote =
+        get_proportion_i128(per_lp_delta_quote, user_lp_shares, AMM_RESERVE_PRECISION)?;
 
-        (lp_delta_base, lp_delta_quote, lp_fee)
-    } else {
-        (0, 0, 0)
+    let per_lp_position_delta = PositionDelta {
+        base_asset_amount: -per_lp_delta_base,
+        quote_asset_amount: -per_lp_delta_quote,
     };
 
-    let amm_fee = fee_to_market
-        .checked_sub(lp_fee)
-        .ok_or_else(math_error!())?;
+    update_amm_position(market, &per_lp_position_delta, true)?;
 
-    // Update AMM position
-    let amm_baa = delta
-        .base_asset_amount
-        .checked_sub(lp_delta_base)
+    // 1/5 of fee auto goes to market
+    // the rest goes to lps/market proportional
+    let lp_fee = (fee_to_market - (fee_to_market / 5)) // todo: 80% retained
```

Comment:
lets add an LP fee numerator/denominator as well in constants? (just so its not hardcoded)